thrust/thrust/system/tbb/detail/reduce_intervals.h

```diff
@@ -81,7 +81,7 @@ struct body
     RandomAccessIterator1 my_last  = first + offset_to_last;
 
     // carefully pass the init value for the interval with raw_reference_cast
-    using sum_type = typename BinaryFunction::result_type;
```

Comment:
Given @alliepiper's answer on [this PR](https://github.com/NVIDIA/cccl/pull/1998#issuecomment-2237236354), this may be a breaking change that we have to delay. To make progress with this PR (because I really want the deprecation warnings to come in), please revert this change and surround this line with:
```c++
_CCCL_SUPPRESS_DEPRECATED_PUSH
using sum_type = typename BinaryFunction::result_type;
_CCCL_SUPPRESS_DEPRECATED_POP
```
This gives us more time to find a replacement for this line.

Reply:
#1998 got merged, and your original replacement is the way we should go. Also since `_CCCL_SUPPRESS_DEPRECATED_PUSH` does not seem to work on ICC (I have a workaround locally, but it's ugly and we want to get rid of ICC anyhow). Just going for the proper solution seems the cleanest. I will push the change into your PR.

Reply:
Thanks for updating the PR.